x-pack/plugins/ml/server/lib/alerts/register_anomaly_detection_alert_type.ts

```diff
@@ -42,10 +62,88 @@ export type AnomalyDetectionAlertContext = AnomalyDetectionAlertBaseContext & {
   topInfluencers?: InfluencerAnomalyAlertDoc[];
 };
 
+export type ExecutorOptions<P extends RuleTypeParams> = RuleExecutorOptions<
+  P,
+  RuleTypeState,
+  {},
+  AnomalyDetectionAlertContext,
+  typeof ANOMALY_SCORE_MATCH_GROUP_ID,
+  MlAnomalyDetectionAlert
+>;
+
 export const ANOMALY_SCORE_MATCH_GROUP_ID = 'anomaly_score_match';
 
 export type AnomalyScoreMatchGroupId = typeof ANOMALY_SCORE_MATCH_GROUP_ID;
 
+export const ANOMALY_DETECTION_AAD_INDEX_NAME = 'ml.anomaly-detection';
+
+const ML_ALERT_NAMESPACE = ALERT_NAMESPACE;
+
+export const ALERT_ANOMALY_DETECTION_JOB_ID = `${ML_ALERT_NAMESPACE}.job_id` as const;
+
+export const ALERT_ANOMALY_SCORE = `${ML_ALERT_NAMESPACE}.anomaly_score` as const;
+export const ALERT_ANOMALY_IS_INTERIM = `${ML_ALERT_NAMESPACE}.is_interim` as const;
+export const ALERT_ANOMALY_TIMESTAMP = `${ML_ALERT_NAMESPACE}.anomaly_timestamp` as const;
+
+export const ALERT_TOP_RECORDS = `${ML_ALERT_NAMESPACE}.top_records` as const;
+export const ALERT_TOP_INFLUENCERS = `${ML_ALERT_NAMESPACE}.top_influencers` as const;
+
+export const ANOMALY_DETECTION_AAD_CONFIG: IRuleTypeAlerts<MlAnomalyDetectionAlert> = {
+  context: ANOMALY_DETECTION_AAD_INDEX_NAME,
+  mappings: {
+    fieldMap: {
+      [ALERT_ANOMALY_DETECTION_JOB_ID]: {
+        type: ES_FIELD_TYPES.KEYWORD,
+        array: false,
+        required: true,
+      },
+      [ALERT_ANOMALY_SCORE]: { type: ES_FIELD_TYPES.DOUBLE, array: false, required: false },
+      [ALERT_ANOMALY_IS_INTERIM]: { type: ES_FIELD_TYPES.BOOLEAN, array: false, required: false },
+      [ALERT_ANOMALY_TIMESTAMP]: { type: ES_FIELD_TYPES.DATE, array: false, required: false },
+      [ALERT_TOP_RECORDS]: {
+        type: ES_FIELD_TYPES.NESTED,
+        array: true,
+        required: false,
+        dynamic: false,
+        properties: {
+          job_id: { type: ES_FIELD_TYPES.KEYWORD },
+          record_score: { type: ES_FIELD_TYPES.DOUBLE },
+          initial_record_score: { type: ES_FIELD_TYPES.DOUBLE },
+          detector_index: { type: ES_FIELD_TYPES.INTEGER },
+          is_interim: { type: ES_FIELD_TYPES.BOOLEAN },
+          timestamp: { type: ES_FIELD_TYPES.DATE },
+          partition_field_name: { type: ES_FIELD_TYPES.KEYWORD },
+          partition_field_value: { type: ES_FIELD_TYPES.KEYWORD },
+          over_field_name: { type: ES_FIELD_TYPES.KEYWORD },
+          over_field_value: { type: ES_FIELD_TYPES.KEYWORD },
+          by_field_name: { type: ES_FIELD_TYPES.KEYWORD },
+          by_field_value: { type: ES_FIELD_TYPES.KEYWORD },
+          function: { type: ES_FIELD_TYPES.KEYWORD },
+          typical: { type: ES_FIELD_TYPES.DOUBLE },
+          actual: { type: ES_FIELD_TYPES.DOUBLE },
+          field_name: { type: ES_FIELD_TYPES.KEYWORD },
+        },
+      },
+      [ALERT_TOP_INFLUENCERS]: {
+        type: ES_FIELD_TYPES.NESTED,
```

Comment:
@XavierM Does the alerts table support KQL queries against nested fields? I don't believe we have any other nested fields in the AAD indices as of now.